sections/git_branch.zsh

```diff
@@ -25,6 +37,25 @@ spaceship_git_branch() {
   git_current_branch="${git_current_branch#heads/}"
   git_current_branch="${git_current_branch/.../}"
 
+  # Build commit SHA info if enabled
+  if [[ $SPACESHIP_GIT_BRANCH_SHOW_COMMIT == true ]]; then
+    local git_commit_prefix="$SPACESHIP_GIT_BRANCH_COMMIT_PREFIX$SPACESHIP_GIT_BRANCH_COMMIT_SYMBOL"
+    local git_commit_sha=$(git rev-parse --short HEAD 2> /dev/null)
+    local git_commit_info="$git_commit_prefix$git_commit_sha$SPACESHIP_GIT_BRANCH_COMMIT_SUFFIX"
+  fi
+
+  git_current_branch+="$git_commit_info"
```

Comment:
Purely a style thing:

If SPACESHIP_GIT_BRANCH_SHOW_COMMIT isn't `true`, `git_commit_info` won't be set. zsh doesn't mind —you won't get an error— but in theory it would be cleaner to make sure the variable is declared. Something like

```shell
local git_commit_info git_current_branch="$vcs_info_msg_0_"
…
if [[ $SPACESHIP_GIT_BRANCH_SHOW_COMMIT == true ]]; then
  …
  git_commit_info=…
fi
…
```

If you choose to make that change, you could do it with `git_tag_info` too, declaring all three on line 34